pandas/core/algorithms.py

```diff
@@ -1679,8 +1680,22 @@ def map_array(
         return arr.copy()
 
     # we must convert to python types
-    values = arr.astype(object, copy=False)
+    values_as_object = arr.astype(object, copy=False)
     if na_action is None:
-        return lib.map_infer(values, mapper)
+        if isinstance(arr.dtype, BaseMaskedDtype):
+            arr = cast("BaseMaskedArray", arr)
+            if arr._hasna and arr.dtype.na_value is libmissing.NA:
```

Comment:
```suggestion
            if arr._hasna:
```

Is this even required? Wouldn't the subsequent `arr.dtype.na_value` used as an argument already get us what we need?

Reply:
Thanks : You're right